src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeWriter.cs

```diff
@@ -316,5 +339,49 @@ private async ValueTask<FlushResult> FlushAsyncInternal(CancellationToken cancel
                 }
             }
         }
+
+        private void FlushInternal()
+        {
+            // Write all completed segments and whatever remains in the current segment
+            // and flush the result.
+            if (_bytesBuffered > 0)
+            {
+                // Update any buffered data
+                _tail.End += _tailBytesBuffered;
+                _tailBytesBuffered = 0;
+            }
+
+            BufferSegment segment = _head;
+            while (segment != null)
+            {
+                BufferSegment returnSegment = segment;
+                segment = segment.NextSegment;
+
+                if (returnSegment.Length > 0)
+                {
+#if netcoreapp
+                    InnerStream.Write(returnSegment.Memory.Span);
```

Comment:
Is it really better to use the Span APIs here?

Reply:
We have a memory so the Span overload is the most logical.

Reply:
Nit: might be cleaner to move the ifdef into the Stream.Write(ROM) overload.